tensorflow/python/framework/tensor_shape.py

```diff
@@ -329,18 +379,18 @@ def __add__(self, other):
       return Dimension(self._value + other.value)
 
   def __radd__(self, other):
-    """Returns the sum of `other` and `self`.
+    """Sum two known dimension values.
 
     Args:
-      other: Another Dimension, or a value accepted by `as_dimension`.
+      other: Another Dimension.
 
     Returns:
       A Dimension whose value is the sum of `self` and `other`.
     """
     return self + other
 
   def __sub__(self, other):
-    """Returns the subtraction of `other` from `self`.
+    """Subtraction of `other` from `self`.
```

Comment:
This actually gives user the wrong impression that the value of self is being updated, which is not in this case. The method only return the subtracted value without updating self and other.

Reply:
same of all the methods above and below.